video/out/wayland_common.c

```diff
@@ -2604,12 +2604,12 @@ void vo_wayland_wait_frame(struct vo_wayland_state *wl)
     wl->timeout_count = 0;
 }
 
-void vo_wayland_wait_events(struct vo *vo, int64_t until_time_us)
+void vo_wayland_wait_events(struct vo *vo, int64_t until_time_ns)
 {
     struct vo_wayland_state *wl = vo->wl;
 
-    int64_t wait_us = until_time_us - mp_time_us();
-    int timeout_ms = MPCLAMP((wait_us + 999) / 1000, 0, 10000);
+    int64_t wait_ns = until_time_ns - mp_time_ns();
+    int timeout_ms = MPCLAMP((wait_ns + 999000) / 1e6, 0, 10000);
```

Comment:
Should be `999999`. But why do we even round up the sleep time? Is there good reason for it, or just because?

Reply:
Probably something cargoculted from the distant past I'm guessing.

Reply:
Ah avoiding a 0 wait apparently according to this https://github.com/mpv-player/mpv/commit/37d6604d70c8c594de2817db26356c4c950ac0fd. Could have just clamped it to 1 though? Also, I just noticed that drm was never changed so it still uses 500.